types/mydecimal.go

```diff
@@ -713,9 +713,6 @@ func (d *MyDecimal) doMiniRightShift(shift, beg, end int) {
 // RETURN VALUE
 //  eDecOK/eDecTruncated
 func (d *MyDecimal) Round(to *MyDecimal, frac int, roundMode RoundMode) (err error) {
-	if frac > mysql.MaxDecimalScale {
-		frac = mysql.MaxDecimalScale
-	}
```

Comment:
https://github.com/mysql/mysql-server/blob/ba1a99c5cd7dd6204abbe8d177c7f71a48e53fe3/strings/decimal.c#L1641 In port source doesn't check it too.